media-libs/icclib: update EAPI 7 -> 8, fix build problems

NHOrus opened the pull request:
Update code to compile with C23 compilers.
Replace build system with meson: existing makefile and patches appear to have silently dropped dependencies, that were automagically included only somewhere. Fixes segfaults on start.

Closes: https://bugs.gentoo.org/875098
Closes: https://bugs.gentoo.org/456780

<!-- Please put the pull request description above -->

---

Please check all the boxes that apply:

- [X] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [X] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [X] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [X] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

NHOrus committed:
media-libs/icclib: update EAPI 7 -> 8, fix build problems

Update code to compile with C23 compilers.
Replace build system with meson: existing makefile and patches appear
to have silently dropped dependencies, that were automagically included
only somewhere. Fixes segfaults on start.
Adds smoke test to check for said segfault

Closes: https://bugs.gentoo.org/875098
Closes: https://bugs.gentoo.org/456780
Signed-off-by: NHOrus <jy6x2b32pie9@yahoo.com>

thesamesam commented:
Could we have a smoketest for https://bugs.gentoo.org/456780?

NHOrus commented:
It wants an actual file to work on, and prints help with status 1 on any flag it doesn't recognize. I'm not sure how to wire it correctly into smoke test.

thesamesam commented:
`The crash happens when the function usage() is being called.`

We can do something like:
```
./x &| tee log
if ! grep -q "Unrecognised" log ; then
die "Executable couldn't be started"
fi
```

NHOrus committed:
media-libs/icclib: add 3.2.0

Upstream fixed bugs we patch in, and also included test files
which simplifies running smoke test significantly.

Signed-off-by: NHOrus <jy6x2b32pie9@yahoo.com>
